packages/dd-trace/src/remote_config/scheduler.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+class Scheduler {
+  constructor (callback, interval) {
+    this._timer = null
+    this._callback = callback
+    this._interval = interval
+    this._running = false
+  }
+
+  func () {
+    if (this._running) return
+    this._running = true
+
+    this._callback(() => {
+      this._running = false
+    })
+  }
+
+  start () {
+    if (this._timer) return
+
+    const cb = () => this.func()
+
+    setImmediate(cb)
+
+    this._timer = setInterval(cb, this._interval)
```

Comment:
Using an interval can be dangerous for HTTP requests if the timeout is higher than the interval. It's safer to instead schedule the next call after the previous call is done.

Reply:
I know and that's why I added a path that checks if the request is still running. If the callback hasn't been called when the new interval kicks in, it just skips this iteration.
I've never liked to do setTimeouts in a loop but I don't remember why. It just feels weird to me. You think I should implement it with setTimeouts instead ?

Reply:
`setInterval` is better suited when something should happen every X amount of time exactly. In our case, it's better that a new timer starts after the operation is finished to avoid any overlaps. Basically we wait for X time after each iteration instead of from the previous schedule. It's also worth noting that for long-polling this would basically be required since a new poll should always happen right after the previous one has finished.